lib/system.nim

```diff
@@ -2702,6 +2702,7 @@ when hasAlloc and notJSnotNims:
 
   proc deepCopy*[T](y: T): T =
     ## Convenience wrapper around `deepCopy` overload.
+    result = default(T)
     deepCopy(result, y)
```

Comment:
Maybe `deepCopy(var T, T)` above can become `deepCopy(out T, T)`? Or `mDeepCopy` can be specialized in the compiler like `mNew` [here](https://github.com/nim-lang/Nim/blob/aeb3fe9505f7d1ef3efcd0001541a64ca8382b01/compiler/sempass2.nim#L1044) to emulate `out T`